internal/storage/postgres/tree/create.go

```diff
@@ -39,16 +39,28 @@ func (r *TreeRepository) Create(ctx context.Context, tFn ...entities.EntityFunc[
 }
 
 func (r *TreeRepository) CreateAndLinkImages(ctx context.Context, tFn ...entities.EntityFunc[entities.Tree]) (*entities.Tree, error) {
-	entity, err := r.Create(ctx, tFn...)
+	createdEntity, err := r.Create(ctx, tFn...)
 	if err != nil {
 		return nil, err
 	}
 
-	if err := r.handleImages(ctx, entity.ID, entity.Images); err != nil {
-		return nil, err
+	entity := defaultTree()
+	for _, fn := range tFn {
+		fn(&entity)
+	}
```

Comment:
Here is missing a validation. We need to check if specific fields are given in the request otherwise it should return an error. 

For example: For a tree is the longitude and the latitude required. We shouldnot use the demo data for this. This also has to be tested in a new case.

Reply:
I think it would be better to ask @choffmann about this topic, as he is the one who implemented the function. 

However, in my opinion, we don't need to validate the tree attributes again in the repo layer because they are already validated in the service layer. Adding validation in the repo would create redundant checks. 

please have a look on the create function in service tree: 

```
func (s *TreeService) Create(ctx context.Context, treeCreate *entities.TreeCreate) (*entities.Tree, error) {
    if err := s.validator.Struct(treeCreate); err != nil {
        return nil, service.NewError(service.BadRequest, errors.Wrap(err, "validation error").Error())
    }
```
the role of the repo is to handle database interactions while the service layer is reponsible for business logic and validations.

Reply:
You can also take a look here: [https://github.com/green-ecolution/green-ecolution-backend/blob/ec2668316ad13a93c59bc12674cf78aa44270515/internal/storage/postgres/treecluster/create.go#L35](https://github.com/green-ecolution/green-ecolution-backend/blob/ec2668316ad13a93c59bc12674cf78aa44270515/internal/storage/postgres/treecluster/create.go#L35)

This is the treecluster repository which also Cedrik wrote. He is validating everything again. We should unify this!

Reply:
has been discussed.